webapp/src/client/client.js

```diff
@@ -1,15 +1,11 @@
 import {Client4} from 'mattermost-redux/client';
 import {ClientError} from 'mattermost-redux/client/client4';
 
-import {id as pluginId} from '../manifest';
+import {getSiteURL} from '../selectors';
 
 export default class Client {
-    constructor() {
-        this.url = `/plugins/${pluginId}/api/v1`;
-    }
-
-    getAttributes = async (userID = '') => {
-        return this.doGet(`${this.url}/attributes?user_id=` + userID);
+    getAttributes = async (state = {}, userID = '') => {
```

Comment:
Seems a bit weird to have this function explicitly take the whole state as one of its arguments. Can we instead set the `Client`'s server URL on plugin initialization, similar to how it's done in the zoom plugin? https://github.com/mattermost/mattermost-plugin-zoom/blob/0ba667b10ac437d63600686753b985476eb9bb95/webapp/src/index.js#L26

Reply:
Thank you for pointing to the right direction! Addressing this in the next commit.